From c4cd8b311e84b830f41d7f0e5fc8f49858f5232f Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Wed, 20 Jun 2018 20:01:06 -0700
Subject: [PATCH 02/10] dm: video: bridge: don't fail to activate bridge if
 sleep gpio is missing

Sleep gpio is optional, so it's possible to have reset gpio, but no sleep gpio.
We shouldn't fail early in case of missing sleep gpio, otherwise we won't
deassert reset.

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 drivers/video/bridge/video-bridge-uclass.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/bridge/video-bridge-uclass.c b/drivers/video/bridge/video-bridge-uclass.c
index cd4959cc71..12ae9df5d9 100644
--- a/drivers/video/bridge/video-bridge-uclass.c
+++ b/drivers/video/bridge/video-bridge-uclass.c
@@ -110,7 +110,7 @@ int video_bridge_set_active(struct udevice *dev, bool active)
 
 	debug("%s: %d\n", __func__, active);
 	ret = dm_gpio_set_value(&uc_priv->sleep, !active);
-	if (ret)
+	if (ret && ret != -ENOENT)
 		return ret;
 	if (active) {
 		ret = dm_gpio_set_value(&uc_priv->reset, true);
@@ -120,7 +120,7 @@ int video_bridge_set_active(struct udevice *dev, bool active)
 		ret = dm_gpio_set_value(&uc_priv->reset, false);
 	}
 
-	return ret;
+	return 0;
 }
 
 UCLASS_DRIVER(video_bridge) = {
-- 
2.18.1

